### ISSUE TITLE
Exceptions logged during RSpec controller tests

### ISSUE THREAD

**Issue by apmiller108** [DataDog/dd-trace-rb#2251]
**Current behaviour**
Exceptions are raised, rescued and logged when running RSpec controller tests that use [anonymous controllers](https://relishapp.com/rspec/rspec-rails/docs/controller-specs/anonymous-controller). Tracing is not enabled in the test environment; however, it is my understanding that instrumentation will still run.

```
E, [2022-08-31T13:16:50.455100 #26632] ERROR -- ddtrace: [ddtrace] (/usr/local/bundle/ruby/2.7.0/gems/ddtrace-1.1.0/lib/datadog/tracing/contrib/action_pack/action_controller/instrumentation.rb:43:in `rescue in start_processing') undefined method `resource=' for nil:NilClass
Did you mean?  rescue
E, [2022-08-31T13:16:50.456920 #26632] ERROR -- ddtrace: [ddtrace] (/usr/local/bundle/ruby/2.7.0/gems/ddtrace-1.1.0/lib/datadog/tracing/contrib/action_pack/action_controller/instrumentation.rb:79:in `rescue in finish_processing') undefined method `resource=' for nil:NilClass
Did you mean?  rescue
```

**Expected behaviour**
This doesn't cause the tests to fail since the exceptions are rescued; rather just creates noise in the test output since the exceptions are logged. Would it be possible to temporarily disable the Rails instrument as is possible with some of the other instruments; or perhaps some other work around to silence this?

**Steps to reproduce**

With the Rails instrument enabled, run an RSpec controller test with an [anonymous controller](https://relishapp.com/rspec/rspec-rails/docs/controller-specs/anonymous-controller)

```ruby
describe SomeController do
  controller do
    def index
      head :ok
    end
  end

  describe '#index' do
    it 'responds with a 200 status' do
      get :index
      expect(response).to have_http_status :ok
    end
  end
end
```

**How does `ddtrace` help you?**

**Environment**

* **ddtrace version:** 1.1
* **Configuration block (`Datadog.configure ...`):**
```
Datadog.configure do |c|
  if Rails.env.development? ||  Rails.env.test?
    c.tracing.enabled = false # Instrumentation will still run but no traces are sent to the trace agent
  end

  c.tracing.instrument :faraday, split_by_domain: true 
  c.tracing.instrument :http, split_by_domain:  true
  c.tracing.instrument :rails
  c.tracing.instrument :active_support, cache_service: c.service
  c.tracing.instrument :redis, service_name: "#{c.service}-redis"
end

```
* **Ruby version:** 2.7
* **Operating system:** Centos 7
* **Relevant library versions:** rspec 3.7, rspec-rails 3.7.2

**Comment by oqrusk** [comment]
I'm facing the same issue on circleci

 image: cimg/ruby:2.7.5-browsers
    rspec-core (3.10.1)
    ddtrace (1.1.0)

**Comment by marcotc** [comment]
👋, this issue has been fixed in [ddtrace 1.3.0](https://github.com/DataDog/dd-trace-rb/releases/tag/v1.3.0): https://github.com/DataDog/dd-trace-rb/commit/f6813a1e36fd3fe1d420980eecca8e546fde34c8

Let us know if upgrading to this version works for your use cases.

**Comment by apmiller108** [comment]
@marcotc this is no logger an issue after upgrading to v1.4.1. Thanks 👍
